### ISSUE TITLE
Request: Ground Station in VRS

### ISSUE THREAD

**Issue by jaymot123** [szpajder/dumphfdl#23]
Similar to `--freq-as-squawk`, can the ground station be added into a field in VRS that's not used in HFDL? VRS has a User Tag field in the aircraft details that's not being shown or used by default which would be good for this. The Tag field, when this option is added to the aircraft details pane, appears just below Notes. It's also available in the List. It would be very nice to know what ground stations I'm receiving messages from in addition to the frequencies. If used along with the `--system-table` option it could show the "friendly names" of the ground stations. Otherwise, to find this information I'd have to look up the frequency, and in the case of the same freq being used by more than one station I still wouldn't know for sure which one sent the message.

**Comment by szpajder** [comment]
User tag is statically assigned to an airframe in the basestation.sqb database. It is not read off air, hence there is no field in the Basestation format feed that could allow passing its value from dumphfdl to VRS.

**Comment by jaymot123** [comment]
Is there any other field in VRS that could be used? Altitude or speed maybe?

**Comment by szpajder** [comment]
These are numeric fields, VRS would reject text strings there.

Obviously these fields could be used to convey ground station numeric ID,
but would this be of any use? You would just have another set of numbers to
memorize.

OTOH, after nearly a year of playing with HFDL I have pretty much memorized
most frequencies I receive and how they relate to ground stations. So I
don't consider having a GS ID in another field in the GUI to be a huge
improvement. Textual description - yes, maybe, but it's not possible. VRS
is not a generic plotting program, it has been designed with ADS-B in mind,
so you can't throw arbitrary data at it and expect it to do something
reasonable with it.

**Comment by jaymot123** [comment]
Oh well, the numeric ground station ID would be better than nothing. I suppose I could print the systable.conf file and refer to it until I'd memorized the ID numbers and station names, or make a symlink to it on the desktop and open it in a text editor for reference.

**Comment by szpajder** [comment]
Ground station ID would be better than nothing, but freq-as-squawk is even better than ground station ID because it conveys more information, ground station ID included (indirectly). Once you print the systable, you'll have frequencies in front of your eyes as well. And when you reformat and sort it by frequency, looking up GS ID will be equally straightforward.

**Comment by jaymot123** [comment]
That's not always true though. 8912 is used by Molokai, Hawaii and Riverhead, New York. 13276 is used by both of those plus San Fransisco California. 17919 is used by San Fransisco, Molokai, Riverhead, Barrow in Alaska and Agana in Guam.  Knowing which ground stations I'm receiving messages from would give me an idea of how well my SDR and antenna setup are performing on any given day and time. Currently, I have no idea (in VRS) of knowing if I'm only receiving messages on 17919 from Guam which is relatively near to the Philippines or from New York or Alaska which is DX for me. The only way to find out is to stop dumphfdl then run it again without porting its output to VRS then watch the actual messages to see what ground stations they're coming from. Knowing both the frequency _and_ the ground station would give me the clearest idea of how my station's performing from within the VRS web interface.

**Comment by szpajder** [comment]
> Currently, I have no idea (in VRS) of knowing if I'm only receiving messages on 17919 from Guam which is relatively near to the Philippines or from New York or Alaska which is DX for me.

VRS won't tell you that. VRS plots plane positions - and these are carried in messages that are sent TO Guam, New York or Alaska, not received FROM them. If you want to know which stations' uplinks you receive, you have to log data to files constantly and inspect their contents - by eyeballing or crafting a script which summarizes them and reports the results.

> Knowing both the frequency and the ground station would give me the clearest idea of how my station's performing and what the propagation's like from within the VRS web interface.

You can judge station performance from VRS GUI already - just by look at the distribution of the planes on the map. If it shows you an aircraft that is 15 thousand km away from your location, then it's a good DX regardless of which station that plane is currently talking to.

In short, what you want to do is already possible using existing feature set. The feature you are asking for won't help you in any way in this.

**Comment by jaymot123** [comment]
"these are carried in messages that are sent TO Guam, New York or Alaska, not received FROM them."
Oh, I see. I was under the impression for some reason that the messages from aircraft were being relayed or rebroadcast by the ground stations, and it was those relayed GS messages that I was receiving. I didn't realize that I'm receiving messages directly from the aircraft themselves.

That being the case, I agree that my request makes no sense for helping me to glean the information that I was hoping for because the ground stations don't really matter when using dumphfdl along with VRS to track planes on a map, only for monitoring actual message content (i.e. seeing conversations between aircraft and ground stations if that's your interest.)
